app/tests/semaphore_test.cc

```diff
@@ -76,6 +88,43 @@ TEST(SemaphoreTest, TimedWait) {
       0.20 * firebase::internal::kMillisecondsPerSecond);
 }
 
+#if SEMAPHORE_LINUX
+// Tests that Timed Wait handles spurious wakeup (Linux/Android specific).
+TEST(SemaphoreTest, TimedWaitSpuriousWakeupLinux) {
+  // Register a dummy signal handler for SIGUSR1; otherwise, sending SIGUSR1
+  // later on will crash the application.
+  signal(SIGUSR1, [](int signum) -> void {});
+
+  // Start a thread that will send SIGUSR1 to this thread in a few moments.
+  pthread_t main_thread = pthread_self();
+  firebase::Thread signal_sending_thread = firebase::Thread(
+      [](void* arg) {
+        firebase::internal::Sleep(500);
+        pthread_kill(*static_cast<pthread_t*>(arg), SIGUSR1);
+      },
+      &main_thread);
+
+  // Call Semaphore::TimedWait() and keep track of how long it blocks for.
+  firebase::Semaphore sem(0);
+  int64_t start_ms = firebase::internal::GetTimestamp();
+  EXPECT_FALSE(sem.TimedWait(2 * firebase::internal::kMillisecondsPerSecond));
```

Comment:
2 * firebase::internal::kMillisecondsPerSecond should be in a constant in this function, for clarity

Reply:
Done.